src/libsync/lockfilejobs.cpp

```diff
@@ -0,0 +1,199 @@
+    record._lockOwnerId = _userId;
+    record._lockEditorApp = _editorName;
+    record._lockTime = _lockTime;
+    record._lockTimeout = _lockTimeout;
+}
+
+SyncJournalFileRecord LockFileJob::handleReply()
+{
+    const auto xml = reply()->readAll();
+
+    QXmlStreamReader reader(xml);
+
+    while (!reader.atEnd()) {
+        const auto type = reader.readNext();
+        const auto name = reader.name().toString();
+
+        switch (type) {
+        case QXmlStreamReader::TokenType::NoToken:
+        case QXmlStreamReader::TokenType::Invalid:
+        case QXmlStreamReader::TokenType::DTD:
+        case QXmlStreamReader::TokenType::EntityReference:
+        case QXmlStreamReader::TokenType::ProcessingInstruction:
+        case QXmlStreamReader::TokenType::Comment:
+        case QXmlStreamReader::TokenType::StartDocument:
+        case QXmlStreamReader::TokenType::Characters:
+        case QXmlStreamReader::TokenType::EndDocument:
+        case QXmlStreamReader::TokenType::EndElement:
+            break;
+        case QXmlStreamReader::TokenType::StartElement:
+            decodeStartElement(name, reader);
+            break;
+        }
+    }
+
+    SyncJournalFileRecord record;
+
+    if (_lockStatus == SyncFileItem::LockStatus::LockedItem) {
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::UserLock && _userDisplayName.isEmpty()) {
+            return record;
+        }
+
+        if (_lockOwnerType == SyncFileItem::LockOwnerType::AppLock && _editorName.isEmpty()) {
+            return record;
+        }
+
+        if (_userId.isEmpty()) {
+            return record;
+        }
+
+        if (_lockTime <= 0) {
+            return record;
+        }
+
+        if (_lockTimeout <= 0) {
+            return record;
+        }
+    }
+
+    const auto relativePath = path().mid(1);
+    if (_journal->getFileRecord(relativePath, &record) && record.isValid()) {
+        setFileRecordLocked(record);
+        _journal->setFileRecord(record);
+        _journal->commit("lock file job");
+    }
+
+    return record;
+}
+
+void LockFileJob::decodeStartElement(const QString &name,
+                                     QXmlStreamReader &reader)
+{
+    if (name == QStringLiteral("lock")) {
+        const auto valueText = reader.readElementText();
+        if (valueText.isEmpty()) {
+            _lockStatus = SyncFileItem::LockStatus::UnlockedItem;
+        } else {
+            bool success = false;
+            _lockStatus = static_cast<SyncFileItem::LockStatus>(valueText.toInt(&success));
+            if (!success) {
+                _lockStatus = SyncFileItem::LockStatus::UnlockedItem;
+            }
+        }
+    } else if (name == QStringLiteral("lock-owner-type")) {
+        const auto valueText = reader.readElementText();
+        _lockOwnerType = static_cast<SyncFileItem::LockOwnerType>(valueText.toInt());
+    } else if (name == QStringLiteral("lock-owner-displayname")) {
+        const auto valueText = reader.readElementText();
```

Comment:
@mgallien Why do you need a temporary variable `valueText` when just assigning it on the next line?

Reply:
did refactor to address your comment